flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/OperatorConfigOptions.java

```diff
@@ -61,12 +62,18 @@
                     .durationType()
                     .defaultValue(Duration.ofSeconds(10))
                     .withDescription(
-                            "The interval before a savepoint trigger attempt is marked as unsuccessful");
+                            "The interval before a savepoint trigger attempt is marked as unsuccessful.");
 
     public static final ConfigOption<Duration> OPERATOR_OBSERVER_FLINK_CLIENT_TIMEOUT =
             ConfigOptions.key("operator.observer.flink.client.timeout")
                     .durationType()
                     .defaultValue(Duration.ofSeconds(10))
                     .withDescription(
                             "The timeout for the observer to wait the flink rest client to return.");
+
+    public static final ConfigOption<String> OPERATOR_DEPLOYMENT_VALIDATOR_TYPE =
+            ConfigOptions.key("operator.deployment.validator.type")
```

Comment:
I think we should support a comma separated list of custom validators: `operator.validation.custom.validators: MyValidator1, MyValidator2`
And maybe we could always have the default validator applied (by default) and add a config option to disable it.
`operator.validation.default.validator.enabled: true/false`

Reply:
Let's think about this design a little more , looping in @wangyang0918 as well.

Reply:
I actually meant the opposite @Aitozi . I think maybe the validator interface should have one explicit method for each relevant resource type. 

validateDeployment, validateSessionJob

They could have a default empty implementation if we want. This might make it easier to implement custom validators.

Reply:
Hah, I know your meaning now, I think it looks good to me. In this way, we can use one option to specify the custom validator

Reply:
It also has the benefit that the user knows what can be validated and what not :)

Reply:
+1 on the suggested changes by @gyfora

Reply:
> support

At present, the implementation has considered the comma separated list and get corresponding validator by the lower case values of the custom validators.

Reply:
? I dont understand sorry.  The configoption should use `.asList()` and allow the user to pass a list of identifiers.

Reply:
Let meg give you an example to clarify:
The user provides the list: `MyValidator1, MyValidator2`

In this case the validtation would be :
 1. Validate using DefaultValidator -> if error, return
 2. Validate using MyValidator1 -> if error, return
 3. Validate using MyValidator2 -> if error, return
 
If all validators successfully validated, perform reconcile

Reply:
Why do we need this config options? Just like Flink plugin mechanism, the name of each plugin(validator) could be simply same as the folder name.
```
/opt/flink-operator/plugins
/opt/flink-operator/plugins/validator-1
/opt/flink-operator/plugins/validator-2
... ...
```

Reply:
@wangyang0918 I think the config is still necessary to select from the available validators no?

Reply:
So do you mean the scenario that user puts a validator-x in the plugins, but he/she does not want it taken effect?

Reply:
Yea maybe it is better to enable the validators automatically if they are in the plugins. That will simplify a bunch of things, get rid of the identifier, config options etc.

Good point